### ISSUE TITLE
Pyright fails to recognize the last empty line in editor text

### ISSUE THREAD

**Issue by StellaHuang95** [microsoft/pyright#5251]
**Describe the bug**
The current behavior of Pyright causes an issue when there is an empty new line as the last line of the editor text. For example, if the text is "Hello\n", Pyright mistakenly recognizes it as a single line instead of two lines. 

**To Reproduce**
In packages\pyright-internal\src\tests\tokenizer.test.ts, run the following unit test:
```typescript
test('NewLines', () => {
    const t = new Tokenizer();
    const results = t.tokenize('hello\r\n');
    assert.equal(results.lines.count, 2);
});
```
The test fails.

**Comment by erictraut** [comment]
I'm not convinced this is a bug. Tokenized lines end on a `\r`, `\n` or `\r\n`. It's possible for the final line not to be terminated. In the example above, the last line is terminated, but there's only one line. Lines should not have zero length; they should always have at least a termination character. In the example above, there's only one line. If the tokenizer were to emit a second line, it would have zero length.

**Comment by debonte** [comment]
> Lines should not have zero length; they should always have at least a termination character. 

Are you saying that there is code that relies on this rule?

From the editor's perspective there are two lines, and `character` zero of the second line is a valid place for the cursor to be. LSP requests that reference such a position are currently problematic for code that uses `tokenizerOutput.lines`.

**Comment by erictraut** [comment]
I'm saying that the code is working the way it was intended to work, so I don't consider this a failure, a mistake, or a bug.

What I hear you saying is that it would be more convenient for pylance if the tokenizer's behavior changed here. That's reasonable.

I'm not sure if there is code that relies on the current design. I suspect not. Nothing comes to mind. I can review the usage of the `lines` output within the pyright code base. You might want to do the same in the pylance code base.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.314, which I just published. It will also be included in this week's Pylance prerelease build.
